pywren/scripts/standalone.py

```diff
@@ -310,41 +310,57 @@ def server(aws_region, max_run_time, run_dir, sqs_queue_name, max_idle_time,
     time.sleep(rand_sleep)
 
     session = boto3.session.Session(region_name=aws_region)
-
     # make boto quiet locally FIXME is there a better way of doing this?
     logging.getLogger('boto').setLevel(logging.CRITICAL)
     logging.getLogger('boto3').setLevel(logging.CRITICAL)
     logging.getLogger('botocore').setLevel(logging.CRITICAL)
 
-
-    # NOTE : This assumes EC2 but in the future we could run on
-    # millennium if we set the log stream correctly
-    instance = get_my_ec2_instance(aws_region)
-    ec2_metadata = get_my_ec2_meta(instance)
-    server_name = ec2_metadata['Name']
-    log_stream_prefix = ec2_metadata['instance_id']
-
-    log_format_str = '{} %(asctime)s - %(name)s- %(levelname)s - %(message)s'\
-                     .format(server_name)
-
-    formatter = logging.Formatter(log_format_str, "%Y-%m-%d %H:%M:%S")
-
-
-    handler = watchtower.CloudWatchLogHandler(send_interval=20,
-                                              log_group="pywren.standalone",
-                                              stream_name=log_stream_prefix + "-{logger_name}",
-                                              boto3_session=session,
-                                              max_batch_count=10)
-
-    debug_stream_handler = logging.StreamHandler()
-    handler.setFormatter(formatter)
-    logger.addHandler(handler)
-    logger.setLevel(logging.DEBUG)
-
-    wren_log = pywren.wrenhandler.logger
-    wren_log.addHandler(handler)
-
-    wren_log.addHandler(debug_stream_handler)
+    def async_log_setup():
+        ''' None of this stuff should be on the critical path to launching an
+            * instance. Instances should start dequeuing from SQS queue as soon
```

Comment:
So what happens to log messages generated before this comes up ? Will we lose them ? Is that okay to do ? (cc @ericmjonas)

Reply:
Yes. I think its fine because the alternative is that the executor crashes or hangs until it can set up the logger...

Reply:
So what happens to log messages generated while the stream is not ready yet. Do they just get dropped ? And is it safe to set this in parallel ? 

It seems to me this is orthogonal to the multi-executor issue as such. Is that right or does this somehow only showup in this case ?

Reply:
Yeah some messages get dropped. I think its a fine design decision?